src/dataflow/src/render/reduce.rs

```diff
@@ -561,10 +561,20 @@ where
                     // hierarchical aggregations; should that belief be incorrect, we
                     // should certainly revise this implementation.
                     let iter = source.iter().map(|(val, _cnt)| val.iter().next().unwrap());
-                    target.push((row_packer.pack(Some(aggr.eval(iter, &RowArena::new()))), 1));
+                    let arena = RowArena::new();
+                    let result = aggr.eval(iter, &arena);
+
+                    for (val, _) in source.iter() {
+                        let datum = val.iter().next().unwrap();
+                        if datum != result {
+                            target.push((row_packer.pack(Some(datum)), 1));
+                        }
+                    }
```

Comment:
I think we want to be a bit careful here. In particular, I think what we actually want to produce is `source` minus `result`, where this will ignore counts in the `source`. I think we might want something like:
```
target.extend(source.iter().cloned());
target.push((result, -1));
```
This has the positive side that we preserve the counts in `source`, so that they are subtracted off completely.

Does that make sense?

Reply:
yes. i think this explains the weirdness i was seeing